packages/connect/src/createScClient/ScProvider/Health.ts

```diff
@@ -202,12 +201,38 @@ class InnerChecker {
     return JSON.stringify(parsedResponse)
   }
 
-  update = (): void => {
+  update = (startNow: boolean): void => {
+    // If `startNow`, clear `#currentHealthTimeout` so that it is set below.
+    if (startNow && this.#currentHealthTimeout) {
+      clearTimeout(this.#currentHealthTimeout)
+      this.#currentHealthTimeout = null
+    }
+
     if (!this.#currentHealthTimeout) {
-      this.#currentHealthTimeout = setTimeout(() => {
-        this.#currentHealthTimeout = null
-        this.startHealthCheck()
-      }, 10000)
+      this.#currentHealthTimeout = setTimeout(
+        () => {
+          this.#currentHealthTimeout = null
+
+          // No matter what, don't start a health request if there is already one in progress.
+          // This is sane to do because receiving a response to a health request calls `update()`.
+          if (this.#currentHealthCheckId) return
+
+          // Actual request starting.
+          this.#currentHealthCheckId = "health-checker:".concat(
+            this.#nextRequestId.toString(),
+          )
+          this.#nextRequestId += 1
+          this.#requestToSmoldot(
+            JSON.stringify({
+              jsonrpc: "2.0",
+              id: this.#currentHealthCheckId,
+              method: "system_health",
+              params: [],
+            }),
+          )
+        },
+        startNow ? 0 : 10000,
+      )
```

Comment:
There is a small change in behavior with these changes that's making some tests to fail. What's changed is that the `startNow` request used to start synchronously. However, with these changes the request is scheduled in a new macrotask.

Either we update the tests or we make the `startNow` request to start synchronously. I suggest keeping the original behavior:

```suggestion
      const startHealthRequest = () => {
        this.#currentHealthTimeout = null

        // No matter what, don't start a health request if there is already one in progress.
        // This is sane to do because receiving a response to a health request calls `update()`.
        if (this.#currentHealthCheckId) return

        // Actual request starting.
        this.#currentHealthCheckId = "health-checker:".concat(
          this.#nextRequestId.toString(),
        )
        this.#nextRequestId += 1
        this.#requestToSmoldot(
          JSON.stringify({
            jsonrpc: "2.0",
            id: this.#currentHealthCheckId,
            method: "system_health",
            params: [],
          }),
        )
      }
      
      if (startNow) startHealthRequest()
      else this.#currentHealthTimeout = setTimeout(startHealthRequest, 1000)
```

Reply:
I kinda need these changes... So, I will commit the suggestion. I hope that's ok.